### DISCUSSION TITLE
Infrastructure Week - Day 1

### DISCUSSION THREAD

**Discussion by the-maldridge** [void-linux/void-packages#45072] [General]
Feel free to ask followup questions in this thread about [the Monday post of infrastructure week.](https://voidlinux.org/news/2023/07/infra-week-day-1.html)

**Comment by dkwo** [comment]
Thanks for this interesting series, it helped me understand better certain aspects of Void.
Are there any plans to have native builders for aarch64, possible others (all?) as well (risc, ppc)? what are the main obstructions to this?
I understand Void does not accept money donations, so what are the options (if any) for users to contribute towards this goal?

**Reply by the-maldridge** [reply]
We have talked about doing native builders before, and they'd certainly allow us to resolve a number of sticky edge cases.  The primary impediment to doing so is finding a suitable native builder, in a facility with reliable power and data, and making sure that it actually has enough resources to build all packages.  We have some packages that take a surprising amount of resources at link time.

As to the contributions towards this goal, one of the best ways to contribute to improving the build infrastructure is to work on XBPS itself.  Any bugs you can fix there or sharp edges that can be rounded off help with improving the workflows for the build farm.

**Reply by dkwo** [reply]
I see, I guess you mean server-grade ARM boards. Do you think something like the RK3588 could do?
Looking forward to the day when Void will have native aarch64, riscv and ppc builders :)

**Reply by dkwo** [reply]
Btw, it seems Hetzner cloud now offers aarch64 with ampere altra.

**Comment by dkwo** [comment]
have you thought about applying to oregon state university for the aarch64 builders?
https://osuosl.org/services/aarch64/current-projects/#foss-projects
it seems chimera linux and many others do that.
